src/Cedar/Protocol.c

```diff
@@ -5187,7 +5187,7 @@ bool ClientConnect(CONNECTION *c)
 
 		sess->EnableBulkOnRUDP = false;
 		sess->EnableHMacOnBulkOfRUDP = false;
-		if (s != NULL && s->IsRUDPSocket && s->BulkRecvKey != NULL && s->BulkSendKey != NULL)
+		if (s->IsRUDPSocket && s->BulkRecvKey != NULL && s->BulkSendKey != NULL)

```

Comment:
Same here. Why was s != NULL? check removed? There's no way for the compiler to know about the actual value of this variable at compile time.

Reply:
If `s` is NULL, then the members accessed via it must be NULL too.

Reply:
if s is NULL we return from function https://github.com/SoftEtherVPN/SoftEtherVPN/blob/master/src/Cedar/Protocol.c#L4821